__tests__/Guides.unit.test.ts

```diff
@@ -1,10 +1,560 @@
+import * as _ from 'css-to-mat';
+import Gesto from 'gesto';
+import { defaultProps, GUIDE } from '../src/react-guides/consts';
 import Guides from '../src/react-guides/Guides';
-import { GuidesProps } from '../src/react-guides/types';
+import { GuidesProps, GuidesState, LockGuides } from '../src/react-guides/types';
+import Ruler, { RulerProps } from '../src/react-ruler';
+const indexOfDeletedGuide = 2;
+const guidesTest: number[] = [5, -6, 3, -4.2];
+const deletedGuide = guidesTest[indexOfDeletedGuide];
+const selectedGuide: number = guidesTest[1];
+const eventMouse: MouseEvent = new MouseEvent('click');
+
+let isDragging = false;
 
 // Guides
 describe('Guides', () => {
-    it('should be defined', () => {
-        const guide = new Guides({} as Required<GuidesProps>);
-        expect(guide).toBeDefined();
+  let guidesInstance: Guides = createInstance();
+  beforeEach(() => {
+    isDragging = false;
+    guidesInstance = createInstance();
+  });
+
+  it('should be defined', () => {
+    expect(guidesInstance).toBeDefined();
+  });
+
+  it('should get correctly ruler props', () => {
+    expect((guidesInstance as any).rulerProps).not.toHaveProperty('style');
+    expect((guidesInstance as any).rulerProps).toMatchObject({ zoom: 1 });
+  });
+
+  it('should guides be received', () => {
+    guidesInstance.getGuides();
+    expect(guidesInstance.state.guides).toEqual([]);
+  });
+
+  it('should guides be loaded', () => {
+    guidesInstance.loadGuides(guidesTest);
+    expect(guidesInstance.getGuides()).toEqual(guidesTest);
+  });
+
+  it('should clear all guides', () => {
+    guidesInstance.loadGuides(guidesTest);
+    guidesInstance.clearAllGuides();
+    expect(guidesInstance.getGuides()).toEqual([]);
+  });
+
+  it('should scroll guides when guides elements are existing and change style for each element', () => {
```

Comment:
test description should be short and clear, so you can easily determinate about what it is